### DISCUSSION TITLE
Any reason to think these GG and Virtual Boy adapters from Save the Hero are incompatible with v5?

### DISCUSSION THREAD

**Discussion by Weef-Bellington** [sanni/cartreader#754] [General Chat]
I just preordered these [Game Gear](https://savethehero.builders/collections/build-service/products/coming-soon-gg-adapter) and [Virtual Boy](https://savethehero.builders/collections/build-service/products/virtual-boy-adapter-diy-kit-set) adapters from Save the Hero Builders. Looks like the Game Gear adapter goes into the Mega Drive/Genesis slot and the Virtual Boy adapter goes into the SNES/SFC slot.

My question is: any reason to think these adapters are not compatible with v5 of the OSCR? I ask only because I know the Wonderswan adapter PCB Save the Hero sells is only compatible with v3 without modifications.

**Comment by Ancyker** [comment]
I'm not sure, but @SAVE-THE-HERO is on here. If not, you could use my GG adapter that works using the SMS slot. It should work with anything that supports SMS because it maps the GG to the equivalent SMS pins. You can even dump a GG cart as an SMS cart with my adapter using a dumper that doesn't list/support GG but does have SMS.

**Comment by SAVE-THE-HERO** [comment]
I have not tried it, but I believe it works fine with V5.

**Comment by Weef-Bellington** [comment]
Thank you @Ancyker and @SAVE-THE-HERO !

I also sent an email to Save the Hero Builders' email address, and got a response from Tetsuo saying the adapters work with V5. I look forward to trying them out in a couple months!

**Reply by Ancyker** [reply]
If you want to try my adapter, https://github.com/StarshadeRETRO/KiCadFiles/tree/main/GameSystems/Adapters/SMS-to-GG

**Comment by sanni** [comment]
You can imagine a HW4/HW5 like a HW3 where the CLK0, CLK1, EEP switches are constantly set to ON.

For CLK0 and CLK1 "ON" means the CLK and CPU CLK pins on the SNES Cart Slot are connected to the Adafruit Clock Generator instead of the pins PH1(Arduino Pin D16) and PE3(Arduino Pin D5) of the Atmega 2560.
For the EEP switch "ON" means there is a constant 1K pull-up resistor on PH4 (Arduino Pin D7).

This has an effect on some systems and especially adapters that plug into the SNES slot:
- All old HW3 flashrom adapters(not part of this Github anymore) are incompatible with HW4/5   
- WonderSwan V1 adapter is incompatible with HW4/5, [V2 adapter fixes that](https://github.com/sanni/cartreader/pull/337)
- Reading N64 Eeprom needs "clockgen_installed" defined on HW4/5, which is done by default on newer firmware releases
- my old npwriter sketch to unlock and write SF Memory carts manually (not part of this Github anymore) doesn't work anymore on HW4/5

So it was an acceptable loss for not having to care about 3 switches anymore. 

Pro tip: If you have a HW3 and are on a newer firmware release just leave the three switches to ON at all times too (unless for WonderSwan adapter).

To answer your question:
- the GG adapter plugs into the Mega Drive slot and is unaffected by this particular hardware change, so it should work on HW5
- Virtual Boy adapter does plug into the SNES slot but it doesn't use the CLK and CPUCLK pins of the SNES slot(not connected on the adapter PCB) so it should work on HW5

**Reply by SAVE-THE-HERO** [reply]
Let me ask you 3 question🙏
What are old HW3 flashrom adapters?
Can the adapters on this page be used with HW4?
https://github.com/sanni/cartreader/tree/bd3eaa106b6e4cbdffbfa89a3fb5fedf3e028b17/pcb/adapters
If I have a HW4, is there any way to use the HW3 adapter on this page?

**Reply by sanni** [reply]
With old flashrom adapters I mean the following PCBs
- 26L6420 adapter
- 27c322 adapter
- flash adapter

And also the even older:
- 8bit flashrom adapter
- 8bit/16-bit combined flashrom adapter 

Basically every old adapter that is meant to write single flashroms is not compatible with HW4/5. Adapters that look like this:

![image](https://dl.dropboxusercontent.com/s/xbf6sfqxvxqku36/IMG_2251.JPG?dl=1)   

The NGP adapter from that link should be compatible since it doesn't use the changed pins on the SNES slot.
The WonderSwan adapter from that link is not compatible with HW4/5 since it's the V1, [V2 from here](https://github.com/sanni/cartreader/pull/337) works with both HW3 and HW4/5

**Reply by Weef-Bellington** [reply]
Thanks for this response @sanni !
